SweepMe/pysweepme
# Project structure and automation

- Use of pyproject.toml to describe metadata and build information of pysweepme
  (modern way of setup.py/cfg and more)
- Use of tox.ini to orchestrate various tests:
  - ruff (style and linting checks, including comparison to main)
  - mypy (type checking)
  - black (formatting, currently disabled)
  - pytest (unit tests)
    - including a test that checks for accidental FolderManager initialization on pysweepme import
  - build (wheel generation, just to check if it can be built, not used for publishing)
 - Github actions for
   - in Pull Requests:  
     - Running all checks defined in the tox.ini
     - Check if the version number in the `__init__.py` has been adjusted (and does not yet exist on pypi)
     - Check if the completion of the PR will generate a semi-linear history (except for merges 1.5.5->1.5.6 etc.)
   - manually triggered:
     - Creating a pysweepme Build and upload it to (test.)pypi (manually triggered)
